odoo/tools/translate.py

```diff
@@ -720,22 +720,60 @@ def unquote(str):
     """Returns unquoted PO term string, with special PO characters unescaped"""
     return re_escaped_char.sub(_sub_replacement, str[1:-1])
 
-def TranslationFileReader(source, fileformat='po'):
+
+def xid_split(xid: str, module: str) -> tuple[str, str]:
+    split_id = xid.split('.', maxsplit=1)
+    if len(split_id) == 1:
+        return module, split_id[0]
+    return split_id[0], split_id[1]
+
+
+def TranslationFileReader(source, fileformat='po', module=None):
     """ Iterate over translation file to return Odoo translation entries """
     if fileformat == 'csv':
-        return CSVFileReader(source)
+        return CSVFileReader(source, module)
     if fileformat == 'po':
         return PoFileReader(source)
+    if fileformat == 'xml':
+        return XMLFileReader(source, module)
     _logger.info('Bad file format: %s', fileformat)
     raise Exception(_('Bad file format: %s', fileformat))
 
 class CSVFileReader:
-    def __init__(self, source):
+    def __init__(self, source, module=None):
         _reader = codecs.getreader('utf-8')
+        self.module = module
+        if self.module:
+            self.model = os.path.splitext((os.path.basename(source.name)))[0].split('-')[0]
         self.source = csv.DictReader(_reader(source), quotechar='"', delimiter=',')
         self.prev_code_src = ""
 
     def __iter__(self):
+        if self.module:
```

Comment:
Using `module` to indicate that a file is translated seems strange. Can you define at least a property like `is_translation_file` to indicate the behaviour?

Reply:
Sure but then `is_data_file = bool(self.module)` or `is_trznslation_file = not self.module`... I find it very similar to what is there now.

Also, it is mentioned in the do string that providing the module name will consider it as being a data file.

Reply:
It is more explicit in this context than reading the doc further in the file.